### PR TITLE
REFACTOR-#5303: Fix code scanning alert - Unused local variable

### LINKED ISSUE #5303: Fix code scanning alert - Unused local variable
Tracking issue for:
- [ ] https://github.com/modin-project/modin/security/code-scanning/310

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5303  
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/batch/test/test_pipeline.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ def new_col_adder(df, partition_id):
             NotImplementedError,
             match="Dynamic repartitioning is currently only supported for DataFrames with 1 partition.",
         ):
-            new_dfs = pipeline.compute_batch()
+            _ = pipeline.compute_batch()
```
Comment (mvashishtha) [inline modin/experimental/batch/test/test_pipeline.py:412]:
```suggestion
            pipeline.compute_batch()
```

Why assign anything at all?

File: modin/experimental/batch/test/test_pipeline.py [hunk 2]
```diff
@@ -454,7 +454,7 @@ def reducer(dfs):
             NotImplementedError,
             match="Fan out is only supported with DataFrames with 1 partition.",
         ):
-            new_df = pipeline.compute_batch()[0]
+            _ = pipeline.compute_batch()[0]
```
Comment (mvashishtha) [inline modin/experimental/batch/test/test_pipeline.py:457]: ditto

File: modin/pandas/test/dataframe/test_default.py [hunk 3]
```diff
@@ -334,7 +332,7 @@ def test_matmul(data):
 
     # Test when input series index doesn't line up with columns
     with pytest.raises(ValueError):
-        modin_result = modin_df @ pd.Series(np.arange(col_len))
+        _ = modin_df @ pd.Series(np.arange(col_len))
```
Comment (mvashishtha) [inline modin/pandas/test/dataframe/test_default.py:335]: ditto

File: modin/pandas/test/test_general.py [hunk 4]
```diff
@@ -308,7 +308,7 @@ def test_merge_asof_suffixes():
             suffixes=(False, False),
         )
     with pytest.raises(ValueError), warns_that_defaulting_to_pandas():
-        modin_merged = pd.merge_asof(
+        _ = pd.merge_asof(
```
Comment (mvashishtha) [inline modin/pandas/test/test_general.py:311]: ditto

File: modin/pandas/test/test_series.py [hunk 5]
```diff
@@ -1080,20 +1080,17 @@ def test_astype(data):
...
 
 
-def test_astype_categorical():
-    modin_df = pd.Series(["A", "A", "B", "B", "A"])
-    pandas_df = pandas.Series(["A", "A", "B", "B", "A"])
+@pytest.mark.parametrize(
+    "data", [["A", "A", "B", "B", "A"], [1, 1, 2, 1, 2, 2, 3, 1, 2, 1, 2]]
+)
+def test_astype_categorical(data):
+    modin_df, pandas_df = pd.Series(data), pandas.Series(data)
```
Comment (mvashishtha) [inline modin/pandas/test/test_series.py:1087]:
```suggestion
    modin_df, pandas_df = create_test_series(data)
```

File: modin/pandas/test/test_series.py [hunk 6]
```diff
@@ -1580,7 +1577,7 @@ def test_matmul(data):
 
     # Test when input series index doesn't line up with columns
     with pytest.raises(ValueError):
-        modin_result = modin_series @ pd.Series(
+        _ = modin_series @ pd.Series(
```
Comment (mvashishtha) [inline modin/pandas/test/test_series.py:1580]: ditto-- why assign? here and below

**Review by mvashishtha — COMMENTED** [review]
Thanks @anmyachev ; I have some minor comments

File: modin/pandas/test/test_series.py (hunk 6 above)
Comment (anmyachev) [reply modin/pandas/test/test_series.py]: Thought it was better for the linter, but no, he doesn't care. fixed

**Review by mvashishtha — APPROVED** [review]

**Comment by anmyachev** [comment]
@mvashishtha let's merge

**Pull request merged**